src/Featurizers/Components/GlobalMinimumFrequencyEstimator.h

```diff
@@ -0,0 +1,222 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include <limits>
+
+#include "TrainingOnlyEstimatorImpl.h"
+#include "FrequencyEstimator.h"
+#include "GrainFeaturizerImpl.h"
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+namespace Components {
+
+// TODO: This should be removed in favor of the StatisticalMetricsEstimator
+
+static constexpr char const * const         GlobalMinimumFrequencyEstimatorName("GlobalMinimumFrequencyEstimator");
+static constexpr char const * const         GrainedFrequencyEstimatorName("GrainedFrequencyEstimator");
+
+using GrainedFrequencyEstimator = GrainEstimatorImpl<std::string, FrequencyEstimator<>>;

```

Comment:
>using GrainedFrequencyEstimator [](start = 0, length = 31)

This using statement should be a template with the MaxNumTrainingItemsV set to std::numeric_limits<size_t>::max() as a default. That template value should be provided to FrequencyEstimator.